src/main/java/com/digitalasset/testing/ledger/SandboxRunner.java

```diff
@@ -24,17 +23,17 @@
   private final boolean useWallclockTime;
   private final Optional<String> ledgerId;
   private final Optional<LogLevel> logLevel;
-  private final Path projectRoot;
+  private final File damlRoot;
   private Process sandbox;
 
-  public SandboxRunner(
-      Path projectRoot,
+  SandboxRunner(
+      Path damlRoot,
       Path relativeDarPath,
       Integer sandboxPort,
       boolean useWallclockTime,
       Optional<String> ledgerId,
       Optional<LogLevel> logLevel) {
-    this.projectRoot = projectRoot;
+    this.damlRoot = damlRoot != null ? damlRoot.toFile() : null;
```

Comment:
minor, feel free to ignore, as this is just my biased opinion: sounds to me like an Optional map, so I would have used `Optional<Path>` and `Optional<File>` here, but I'm okay with this.

Reply:
I thought of that too. But that's unnecessary heap allocation. To me this is clear too. I could wrap this logic into a method as well. Also I was thinking to move this logic where this is being used. So I was thinking:

```java
new ProcessBuilder(getDamlSandboxStarterCommand()).directory(getWorkingDirectory(damlRoot))

private File getWorkingDirectory(Path path) {
  return path != null ? path.toFile() : null;
}